test/CancelOffer.t.sol

```diff
@@ -89,4 +88,25 @@ contract CancelOfferTest is BaseTest {
         // validate that the receiver items are not in escrow
         assertOfferItemsOwnership(offer.receiverItems.items, offer.receiver);
     }
+
+    function testCanCancelOfferMultiTokens() public {
+        Offer memory offer = _createMultiTokensOffer();
+
+        bytes32 offerId = generateOfferId(offer);
+
+        vm.prank(account1);
+        vm.expectEmit(true, true, true, true, address(echo));
+        emit OfferCanceled(offerId);
+        echo.cancelOffer(offerId);
+
+        (address sender,,,,,) = echo.offers(offerId);
+
+        // @dev When the offer is cancelled, it's also deleted
+        assertEq(sender, address(0));
+
+        // validate that the sender items are not in escrow
+        assertOfferItemsOwnership(offer.senderItems.items, offer.sender);
+        // validate that the receiver items are not in escrow
+        assertOfferItemsOwnership(offer.receiverItems.items, offer.receiver);
```

Comment:
shouldn't we assert the offer state as well in these tests?

Reply:
The offer is deleted once it's canceled, so we do check the offer state by checking that the offer actually doesn't exist:

```
(address sender,,,,,) = echo.offers(offerId);

        // @dev When the offer is cancelled, it's also deleted
        assertEq(sender, address(0));
```

This basically does that.